File: Core/src/main/java/com/hqq/core/net/ok/OkHttp.java

package com.hqq.core.net.ok;

/**
 * @Author : huangqiqiang
 * @Package : com.qq.readbook.net.ok
 * @FileName :   OkHttp
 * @Date : 2016/12/14 15:51
 * @Email :  qiqiang213@gmail.com
 * @Descrive :
 */
public class OkHttp {
    private static HttpCompat mOkHttp = null;

    public synchronized static HttpCompat newHttpCompat() {
        if (mOkHttp == null) {
            mOkHttp = new OkHttpImpl().create();
        }
        return mOkHttp;
    }

    /**
     * 空的对象
     *
     * @return
     */
    public static HttpCompat.ParamsCompat newParamsCompat() {
        return new OkParamsImpl();
    }

    /**
     * key value 对象
     *
     * @param key
     * @param values
     * @return
     */
    public static HttpCompat.ParamsCompat newParamsCompat(String key, Object values) {
        HttpCompat.ParamsCompat paramsCompat = newParamsCompat();
        paramsCompat.put(key, values);
        return paramsCompat;
    }
}

File: Core/src/main/java/com/hqq/core/net/ok/OkHttp.kt
package com.hqq.core.net.ok

import com.hqq.core.net.ok.HttpCompat.ParamsCompat

/**
 * @Author : huangqiqiang
 * @Package : com.qq.readbook.net.ok
 * @FileName :   OkHttp
 * @Date : 2016/12/14 15:51
 * @Email :  qiqiang213@gmail.com
 * @Descrive :
 */
object OkHttp {
    private val mOkHttp: HttpCompat by lazy {
        OkHttpImpl().create()
    }

    @Synchronized
    fun newHttpCompat(): HttpCompat {
        return mOkHttp
    }

    /**
     * 空的对象
     *
     * @return
     */
    fun newParamsCompat(): ParamsCompat {
        return OkParamsImpl()
    }

    /**
     * key value 对象
     *
     * @param key
     * @param values
     * @return
     */
    fun newParamsCompat(key: String?, values: Any?): ParamsCompat {
        val paramsCompat = newParamsCompat()
        paramsCompat.put(key, values)
        return paramsCompat
    }
}